### PR TITLE
Fix all non-bug lints. Switch to `let` statements.

### PR DESCRIPTION
**Pull request by JohnHeitmann**
After these changes there is only one error report remaining from the linter, and that appears to be a real bug.

The first commit has all the non-trivial changes. The second is a mechanical s/var/let/.

cc: #51735

### CODE REVIEW

**Comment by JohnHeitmann** [comment]
Ah, right after submit I found it. Unique DOM ids are automatically added to window.

**Comment by Centril** [comment]
@JohnHeitmann Please avoid merge commits; we'd prefer rebasing instead. :)

**Comment by JohnHeitmann** [comment]
Sorry about that. I force-pushed a clean rebase. I kept the two separate commits to keep the more interesting changes apart from the search and replace change.

File: src/librustdoc/html/static/storage.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // From rust:
 /* global resourcesSuffix */
 
+// eslint doesn't understand the functions here are used elsewhere
+/*eslint no-unused-vars: "off"*/
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/storage.js]: Can you add spaces to pad this statement, so it can match the other lint statements?

**Comment by QuietMisdreavus** [comment]
cc @GuillaumeGomez for changes to JS

Do we run eslint in CI? Is it particularly time-intensive? @rust-lang/infra Would it be possible to slot it into one of our existing builders?

**Comment by kennytm** [comment]
Note: https://caniuse.com/#search=let

----

@QuietMisdreavus 

> Do we run eslint in CI?

I can't find anything about `eslint` in this repo 🙃. If we could make it runnable via `./x.py test ...`, we could add this to `mingw-check` or `x86_64-unknown-linux-gnu` (depending on whether a bootstrap is needed).

**Comment by JohnHeitmann** [comment]
While it runs fast, eslint is a node app. Another thread made it sound like node was a challenging dependency to impose upon some distros, so I assumed this would be a best-effort tool used by humans.

If node is ok in the build after all I could take a look at adding this to CI.

**Comment by kennytm** [comment]
We already depend on nodejs to run the `src/test/rustdoc-js` tests (the test is skipped if nodejs is missing).

**Comment by QuietMisdreavus** [comment]
> Somehow test the travis build. Do I just need to set up my own travis instance and it'll just work? Let me know if there are mines here.

If you need to run a specific travis instance on your PR without trying to merge it, you can edit the travis.yml file to run a specific builder on pull requests as well instead of just the "auto" branch. (I'd recommend also renaming the PR here to add "WIP" to it so bors doesn't accidentally merge the travis configuration change.)

**Comment by JohnHeitmann** [comment]
I just did something horrible wrong in my git workflow on this branch and I'm not confident I can back it out without tagging more unrelated issues. Closing this out to start from a fresh branch.

**Pull request closed without merging**